app/models/video.rb

```diff
@@ -14,11 +17,10 @@ def self.search_by_title(video_title)
     video_title.present? ? where("title ILIKE ?", "%#{video_title}%").order(created_at: :desc) : []
   end
   
-  # If large_cover_url has a non empty string and the file name exists then show the image
-  # else show a placeholder image
+  # If large_cover is present then test if it actually exists. If so then display image, in all other cases disply dummy image
   def display_large_video_image
-    if self.large_cover_url.present?
-      File.exists?(Rails.root.join("public" + self.large_cover_url)) ? self.large_cover_url : "http://dummyimage.com/665x375/000/fff.png&text=No+Preview+Available"
+    if self.large_cover.present?
+      self.large_cover.file.exists? ? self.large_cover_url : "http://dummyimage.com/665x375/000/fff.png&text=No+Preview+Available"
```

Comment:
Hey Tomas, that actually checks if the file exists in my S3 bucket as well as to check if there is an entry. Therefore even if there is an entry in large_cover in the database but it doesn't actually exist (may have been deleted for some reason) in my S3 uploads then the dummy image will still be served rather than that small image missing logo that shows up which doesn't look very good.